js-src/compat/plugins/woo-carrier-agents/checkout-woo-carrier-agents.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Checkout scripts for: WooCommerce Carrier Agents (by Markup.fi).
+ */
+
+(function (root, factory) {
+	if ( typeof define === 'function' && define.amd ) {
+		define([], factory(root));
+	} else if ( typeof exports === 'object' ) {
+		module.exports = factory(root);
+	} else {
+		root.CheckoutWooCarrierAgents = factory(root);
+	}
+})(typeof global !== 'undefined' ? global : this.window || this.global, function (root) {
+
+	'use strict';
+
+	var $ = jQuery;
+	var _hasJQuery = ( $ != null );
+
+	var _hasInitialized = false;
+	var _publicMethods = {};
+	var _settings = {
+		searchFieldSelector: 'input[name="woo-carrier-agents-postcode"]',
+		searchButtonSelector: '#woo-carrier-agents-search-button',
+		previousPostcodeFieldSelector: '#woo_carrier_agents-entered_postcode',
+	};
+	var _enteredPostcode = '';
+
+
+	/**
+	 * METHODS
+	 */
+
+
+
+	/**
+	 * Maybe update postcode field with the previously entered value.
+	 */
+	var maybeUpdatePostcodeField = function() {
+		// Get postcode search input field
+		var searchField = document.querySelector( _settings.searchFieldSelector );
+		var searchButton = document.querySelector( _settings.searchButtonSelector );
+
+		// Maybe update postcode field value
+		if ( _enteredPostcode ) {
+			searchField.setAttribute( 'value', _enteredPostcode );
+		}
```

Comment:
> Uncaught TypeError: Cannot read properties of null (reading 'setAttribute')

When using view mode as radio button or select field, the search field is not present in the page. Before setting the field value, we need to check whether the search field exists:

```js
if ( searchField && _enteredPostcode ) {
	// Do something
}
```